doc/dev/kernel.xml

```diff
@@ -1062,90 +1062,101 @@ These are usually set up during initialization of a kernel module.
-describing a module
+<P/>
+
+Basically each module consists of one <C>.c</C> file and one <C>.h</C> file.
+
+The file <C>gap.c</C> contains the variable <C>InitFuncsBuiltinModules</C>
+which is initialized with a list of functions such as <C>InitInfoInt</C>.
+
+Each such function, when called, returns a data structure describing a module,
+which typically looks as follows:
 <Listing>
 <![CDATA[
 /****************************************************************************
 **
 *F  InitInfoInt() . . . . . . . . . . . . . . . . . . table of init functions
 */
 static StructInitInfo module = {
-    MODULE_BUILTIN,                     /* type                           */
-    "integer",                          /* name                           */
-    0,                                  /* revision entry of c file       */
-    0,                                  /* revision entry of h file       */
-    0,                                  /* version                        */
-    0,                                  /* crc                            */
-    InitKernel,                         /* initKernel                     */
-    InitLibrary,                        /* initLibrary                    */
-    0,                                  /* checkInit                      */
-    0,                                  /* preSave                        */
-    0,                                  /* postSave                       */
-    0                                   /* postRestore                    */
+    .type = MODULE_BUILTIN,
+    .name = "integer",
+    .initKernel = InitKernel,
+    .initLibrary = InitLibrary,
+    .checkInit = 0,
+    .preSave = 0,
+    .postSave = 0,
+    .postRestore = 0
 };
 ]]>
 </Listing>
-<P/> *  The first part is just information, but the functions from
+The first part is just information, but the functions from
 <C>initKernel</C> down are the key to the interface to kernel modules.
  
 </Subsection>
 
 <Subsection Label="Subsect-Kernel-Modules-Interface">
 <Heading>The Kernel Module Interface</Heading>
  
-<P/> *  The general rule is that if a 0 appears, the function is skipped
-<P/> *  When GAP is starting up, it does the following:
-<P/> *  Calls the InitKernel functions of all those modules that have
-one. If any of them return non-zero then it aborts.
-<P/> *  Then calls the InitLibrary functions likewise 
-<P/> *  Then the CheckInit functions
-<P/> *  The other three functions relate to save/load workspace
-<P/> *  preSave functions are called before saving.
-<P/> *  postSave functions (in the reverse order) after saving
-<P/> *  postRestore functions are called after a workspace has been
-loaded to finish linking up the kernel and workspace.
+The general rule is that if a 0 appears, the function is skipped;
+indeed, you may omit entries which are 0.<P/>
+
+When GAP is starting up, it does the following:
+<List>
+<Item>Calls the InitKernel functions of all those modules that have
```

Comment:
Ideally, this and friends should use `<C>InitKernel</C>` like a couple of lines below.